### ISSUE TITLE
if u have problem with cifar10-tutorial

### ISSUE THREAD

**Issue by StarPlatinumDa** [NX-AI/vision-lstm#16]
During training, I met a problem with the line: optim.step(). The messages are as follows:
```
_foreach_addcdiv_() received an invalid combination of arguments - got (list, list, tuple, list), but expected one of:
 * (tuple of Tensors self, tuple of Tensors tensor1, tuple of Tensors tensor2, tuple of Scalars scalars)
      didn't match because some of the arguments have invalid types: (!list of [Parameter, Parameter, Parameter, Parameter, Parameter,
```

I don't know why. But since the entrance of this problem is optim.step(), I think there might be something wrong with it. So I just change the optimizer from AdamW to SGD with no changing of other parameters. And it works, the model is now training.
And my GPU is RTX4060. I hope this can help someone or one can help me figure out why did this problem occur.

**Comment by StarPlatinumDa** [comment]
I tried other settings and found that the problem might come from the schedule learning rate code. I commented them out 
```
for param_group in optim.param_groups:
            param_group["lr"] = lrs[update]
```

and it worked.

**Comment by BenediktAlkin** [comment]
hmm, looks like a strange behavior in the AdamW implementation
what torch/cuda version are you using?

**Comment by StarPlatinumDa** [comment]
> hmm, looks like a strange behavior in the AdamW implementation what torch/cuda version are you using?

The cuda version is 12.4 and the pytorch version is 2.0.1.
